src/secrets/src/lib.rs

```diff
@@ -39,3 +44,29 @@ pub enum SecretOp {
         id: GlobalId,
     },
 }
+
+pub struct SecretsReaderConfig {
+    pub mount_path: PathBuf,
+}
+
+pub struct SecretsReader {
+    config: SecretsReaderConfig,
+}
+
+impl SecretsReader {
+    pub fn new(config: SecretsReaderConfig) -> Self {
+        Self { config }
+    }
+
+    #[allow(dead_code)]
+    fn read(&self, id: GlobalId) -> Result<Vec<u8>, anyhow::Error> {
```

Comment:
This should be `pub`. Then you won't need to allow it as dead code.

Reply:
thanks!